src/core/geometry/qgsgeometryutils.h

```diff
@@ -620,6 +620,40 @@ class CORE_EXPORT QgsGeometryUtils
      */
     static QgsLineString perpendicularSegment( const QgsPoint &p, const QgsPoint &s1, const QgsPoint &s2 ) SIP_HOLDGIL;
 
+    /**
+     * \brief Create a perpendicular line segment to a given segment [\a segmentPoint1,\a segmentPoint2] with its center at \a centerPoint.
+     *
+     * May be used to split geometries. The new centered perpendicular line segment will have double the length of the input segment
+     *
+     * \param centerPointx x-coordinate of the point where the center of the perpendicular should be located
+     * \param centerPointy y-coordinate of the point where the center of the perpendicular should be located
+     *
+     * \param segmentPoint1x: x-coordinate of segmentPoint1, the segment's start point
+     * \param segmentPoint1y: y-coordinate of segmentPoint1, the segment's start point
+     * \param segmentPoint2x: x-coordinate of segmentPoint2, the segment's end point
+     * \param y2: y-coordinate of segmentPoint2, the segment's end point
+     *
+     * Result is a line (segment) centered in point p and perpendicular to segment [segmentPoint1, segmentPoint2]
+     *
+     * \param perpendicularSegmentPoint1x: x-coordinate of the perpendicularCenterSegment's start point
+     * \param perpendicularSegmentPoint1y: y-coordinate of the perpendicularCenterSegment's start point
+     * \param perpendicularSegmentPoint2x: x-coordinate of the perpendicularCenterSegment's end point
+     * \param perpendicularSegmentPoint2y: y-coordinate of the perpendicularCenterSegment's end point
+     *
+     * \param (optional parameter) segmentLength trims to given length. Default length is double the length of the input segment. Set to 0 for normalized length which is equal to 1.
+     *
+     *
+     * \since QGIS 3.17?
+     *
+     */
+
+    static void perpendicularCenterSegment(   double centerPointx, double centerPointy,
+                                              double segmentPoint1x, double segmentPoint1y,
+                                              double segmentPoint2x, double segmentPoint2y,
+                                              double &perpendicularSegmentPoint1x SIP_OUT, double &perpendicularSegmentPoint1y SIP_OUT,
+                                              double &perpendicularSegmentPoint2x SIP_OUT, double &perpendicularSegmentPoint2y SIP_OUT,
+                                              double *segmentLength=nullptr
```

Comment:
What do you think about changing this to

    double segmentLength = 0

where a segmentLength of 0 means "default length". If someone wants a normalized segment they can easily set this to 1 then.

Reply:
I think for python scripters it is easier the less arguments there are to fill when there's a function they need to use. Based on that usability perspective I would consider an optional argument more helpful. What do you think?
Good idea! Semantics wise I agree to your proposal.

Reply:
I agree that optional is the way to go. I just find a "three option" argument confusing (nullptr, 0 or value > 0). That's why I'd prefer optional with default as 0 and skip the nullptr option...

Reply:
I understand we don't need two options to achieve the same objective (nullptr, segmentLength=0).
By optional I meant having the choice to add/use the segmentLength argument or not. On one hand I can see an advantage of forcing the script user to actively think about what length to choose for his task. On the other hand I guess the standard use case is probably that the script user will not need to set any length for his task and may not want to care about an additional, irrelevant argument to be set. 
I don't consider this a safety critical function and can see a benefit in user friendliness with the latter option. It can anyhow be confusing as it came to my mind that a segmentLength of 0 would have the same effect as a segmentLength of 2. 
What do you think of keeping the nullptr and eliminating the confusing 0 value option meaning the argument accepts only values >0 ?

Reply:
Using a pointer here is really not ideal, because it makes the function harder to call when you DO want a specific segment length. You'd need to do something like:

    double myLength = 5;
    QgsGeometryUtils::perpendicularCenterSegment( ...... , &myLength );

instead of just

    QgsGeometryUtils::perpendicularCenterSegment( ...... , 5 );

I'm all for making the argument optional, but I'd do that by making the default value 0. ie. the signature becomes:


```
    ...
     * \param segmentLength (optional) Trims to given length. A segmentLength value of 0 refers to the default length which is double the length of the input segment. Set to 1 for a normalized length.
    ...
     */
    ...
    static void perpendicularCenterSegment( double centerPointx, double centerPointy,
    ...
                                            double segmentLength = 0
                                          ) SIP_HOLDGIL;
```

Then it's easy to call by default if you don't care about the segment length, and easy for users to specify a particular desired length (including "1" if they want a normalized segment)

What do you think?

Reply:
Agreed